src/classes/drawings/drawing_commands.vala

```diff
@@ -0,0 +1,170 @@
+/**
+ * Classes to handle drawings over slides.
+ *
+ * This file is part of pdfpc.
+ *
+ * Copyright 2017 Charles Reiss
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+namespace pdfpc.DrawingCommands {
+
+    // A drawing command is a cashed stroke with either a pen or
+    // the eraser.
+    public struct DrawingCommand {
+        bool new_path;
+        bool is_eraser;
+        double x1;
+        double y1;
+        double x2;
+        double y2;
+        float lwidth;
```

Comment:
Is there any reason to use SP floats instead of double? You anyway cast them later to doubles.

Reply:
I was trying to adapt the type to its usage in order to save some space. But in practice it really doesn't matter, since you are not likely have more than a few thousand drawing commands per slide. I'll change all types it to double for consistency.

I just force-pushed again